packages/api/v1/implementation.ts

```diff
@@ -83,6 +86,63 @@ export const ApiContractV1Implementation = createNextRoute(ApiContractV1, {
     }
   }),
 
+  downloadSignedDocument: authenticatedMiddleware(async (args, user, team) => {
+    const { id: documentId } = args.params;
+
+    try {
+      if (process.env.NEXT_PUBLIC_UPLOAD_TRANSPORT !== 's3') {
+        return {
+          status: 500,
+          body: {
+            message: 'Please make sure the storage transport is set to S3.',
+          },
+        };
+      }
+
+      const document = await getDocumentById({
+        id: Number(documentId),
+        userId: user.id,
+        teamId: team?.id,
+      });
+
+      if (
+        !document ||
+        !document.documentDataId ||
+        DocumentDataType.S3_PATH !== document.documentData.type
```

Comment:
This is a 400 error instead since the document exists but we don't support downloading them via the api at the moment.
 We should extract this specific condition into a different return